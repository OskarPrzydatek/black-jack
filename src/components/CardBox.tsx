import { CardColorsEnum } from '../constants/cardColors.enum';
import * as S from '../styles/CardBox.style';
import { CardColorType } from '../types/cardColor.type';
import { CardKindType } from '../types/cardKind.type';

import Icon from './Icon';

interface CardBoxProps {
  kind: CardKindType;
  color: CardColorType;
}

export default function CardBox({ kind, color }: CardBoxProps) {
  const cardKindLabel = () => {
    if (kind === 9 || kind === 10) return kind;
    return kind[0];
  };

  const cardColorStyle = () => {
    if (color === CardColorsEnum.Hearts || color === CardColorsEnum.Tiles)
      return { color: '#D91828' };
    return { color: '#2F2F2F' };
  };

  return (
    <S.CardBox>
      <S.CardLabel>
        <span data-testid="card-box-label" style={cardColorStyle()}>
          {cardKindLabel()}
        </span>
      </S.CardLabel>
      <S.CardColor>
        <Icon cardColor={color} />
      </S.CardColor>
      <S.BottomCardLabel>
        <span style={cardColorStyle()}>{cardKindLabel()}</span>
      </S.BottomCardLabel>
    </S.CardBox>
  );
}
