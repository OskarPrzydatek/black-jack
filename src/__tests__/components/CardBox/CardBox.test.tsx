import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import CardBox from '../../../components/CardBox';
import { CardColorsEnum } from '../../../constants/cardColors.enum';
import { FigureType } from '../../../types/figure.type';

describe('CardBox', () => {
  test('component snapshot', () => {
    const tree = renderer
      .create(<CardBox color={CardColorsEnum.Hearts} kind={9} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure card coloring labels correctly', () => {
    const { rerender } = render(
      <CardBox color={CardColorsEnum.Hearts} kind={9} />
    );
    const label = screen.getByTestId('card-box-label');
    expect(label).toHaveStyle({ color: '#D91828' });
    rerender(<CardBox color={CardColorsEnum.Pikes} kind={9} />);
    expect(label).toHaveStyle({ color: '#2F2F2F' });
  });

  test('ensure card labels works correctly', () => {
    const { rerender } = render(
      <CardBox color={CardColorsEnum.Hearts} kind={9} />
    );
    expect(screen.getByTestId('card-box-label')).toHaveTextContent('9');
    rerender(<CardBox color={CardColorsEnum.Hearts} kind={10} />);
    expect(screen.getByTestId('card-box-label')).toHaveTextContent('10');
    rerender(
      <CardBox color={CardColorsEnum.Hearts} kind={'Jack' as FigureType} />
    );
    expect(screen.getByTestId('card-box-label')).toHaveTextContent('J');
    rerender(
      <CardBox color={CardColorsEnum.Hearts} kind={'Queen' as FigureType} />
    );
    expect(screen.getByTestId('card-box-label')).toHaveTextContent('Q');
    rerender(
      <CardBox color={CardColorsEnum.Hearts} kind={'King' as FigureType} />
    );
    expect(screen.getByTestId('card-box-label')).toHaveTextContent('K');
    rerender(
      <CardBox color={CardColorsEnum.Hearts} kind={'As' as FigureType} />
    );
    expect(screen.getByTestId('card-box-label')).toHaveTextContent('A');
  });
});

export {};
