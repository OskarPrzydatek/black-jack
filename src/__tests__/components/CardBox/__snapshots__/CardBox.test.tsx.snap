// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBox component snapshot 1`] = `
<li
  className="sc-bcXHqe kFDUXW"
>
  <div
    className="sc-gswNZR boNMTz"
  >
    <span
      data-testid="card-box-label"
      style={
        Object {
          "color": "#D91828",
        }
      }
    >
      9
    </span>
  </div>
  <div
    className="sc-hLBbgP biwgZj"
  >
    <svg
      fill="#D91828"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 1c2.21 0 4 1.755 4 3.92C8 2.755 9.79 1 12 1s4 1.755 4 3.92c0 3.263-3.234 4.414-7.608 9.608a.513.513 0 0 1-.784 0C3.234 9.334 0 8.183 0 4.92 0 2.755 1.79 1 4 1z"
      />
    </svg>
  </div>
  <div
    className="sc-gswNZR sc-dkrFOg boNMTz kfyxUn"
  >
    <span
      style={
        Object {
          "color": "#D91828",
        }
      }
    >
      9
    </span>
  </div>
</li>
`;
